internal/querynode/shard_cluster.go

```diff
@@ -210,10 +212,16 @@ func (sc *ShardCluster) addNode(evt nodeEvent) {
 		defer oldNode.client.Stop()
 	}
 
+	client, err := sc.nodeBuilder(evt.nodeID, evt.nodeAddr)
+	if err != nil {
+		// target shall be offline
+		log.Warn("failed to get shardQueryNode", zap.Error(err))
+		return
```

Comment:
The err exception is weird here.

More common way is to call something like GetOrCreateClient in session.go.

Each time we create one client, used the cache if anyone.

Reply:
working on it.

Reply:
This part shall be separated into another pr.